Source/WebCore/platform/graphics/texmap/TextureMapper.cpp

```diff
@@ -169,6 +188,28 @@ Ref<TextureMapperShaderProgram> TextureMapperGLData::getShaderProgram(TextureMap
     return *addResult.iterator->value;
 }
 
+Ref<TextureMapperGPUBuffer> TextureMapperGLData::getBufferFromPool(size_t size, TextureMapperGPUBuffer::Type type)
+{
+    if (!size) {
+        // Use static zero buffer
+        static auto zeroBuffer = TextureMapperGPUBuffer::create(size, type, TextureMapperGPUBuffer::Usage::Dynamic);
+        return zeroBuffer;
+    }
+
+    uint64_t key = (static_cast<uint64_t>(type) << 32) | static_cast<uint32_t>(size);
```

Comment:
You truncate 64bit to 32bit silently here. If size is very big, this function can return a small size buffer.
Let's add RELEASE_ASSERT(size < std::numeric_limits<uint32_t>::max());

Reply:
Added. It's extremely unlikely to happen in practice, even though possible in theory.

To make it fail safe in both 32bit and 64bit cases, `size` could be splitted into two 32bit when `sizeof(size_t) > 4` and refactor key calculation a bit